### DISCUSSION TITLE
Comparing PNG images using ErrorMetric.Absolute option

### DISCUSSION THREAD

**Discussion by bokhanhanna** [dlemstra/Magick.NET#1259] [Help] [answered]
Hello,
I'm trying to compare two png-screenshots using **Magick.NET-Q16-AnyCPU** library (v.12.1.0).
My C# code looks like this:
```
var expectedImagePath = @"C:\Users\Comp\Desktop\expected-image.png";
var actualImagePath = @"C:\Users\Comp\Desktop\actual-image.png";
 
using (var expectedImage = new MagickImage(expectedImagePath))
using (var actualImage = new MagickImage(actualImagePath))
{
    var comparisonResult = expectedImage.Compare(actualImage, ErrorMetric.Absolute);
    Assert.That(comparisonResult == 0);
}
```
Expected image:
---
![expected-image](https://user-images.githubusercontent.com/42610031/191562943-8e788f08-86ea-48ed-a325-35f4c2ce29b6.png)
---
Actual image:
---
![actual-image](https://user-images.githubusercontent.com/42610031/191562937-230e2ca5-2919-4cc9-bb1b-d3938bc017a6.png)
---
The images are visually different, but the result shows they are equal (comparisonResult is "0").

I also tried using another ErrorMetric option:
`var comparisonResult = expectedImage.Compare(actualImage, new ErrorMetric());`
This works as expected and shows that the images are actually different (comparisonResult here is "1").

I'm just wondering why ErrorMetric.Absolute doesn’t work in this situation?

**Comment by dlemstra** [comment, marked as answer]
You will get the correct result when you do `actualImage.Compare(expectedImage` instead. This is happening because the actual image only has two channels and the expected image has 4 channels. This might be a bug that I will need to investigate in more detail. My advise would be to set the `ColorType` of both images to `TrueColorAlpha` before comparing them. Then you are comparing the correct channels. Or you could use `ErrorMetric.RootMeanSquared` instead.

**Reply by bokhanhanna** [reply]
Thanks, this helped!
